mlflow/tracing/export/mlflow_v3.py

```diff
@@ -83,12 +84,21 @@ def _log_trace(self, trace: Trace, prompts: Sequence[PromptVersion]):
         Handles exporting a trace to MLflow using the V3 API and blob storage.
         Steps:
         1. Create the trace in MLflow
-        2. Upload the trace data to blob storage using the returned trace info.
+        2. Log spans using the new log_spans API
+        3. Upload the trace data to blob storage using the returned trace info.
         """
         try:
             if trace:
                 add_size_stats_to_trace_metadata(trace)
                 returned_trace_info = self._client.start_trace(trace.info)
+
+                # Log spans after creating the trace
+                if trace.data.spans:
+                    try:
+                        asyncio.run(self._client.log_spans(trace.data.spans))
+                    except Exception as e:
+                        _logger.debug(f"Failed to log spans: {e}")
```

Comment:
@B-Step62 Here, we update the MLflow V3 exporter to log the spans to the MLflow DB via TracingClient -> RestStore - > OTel REST API (https://github.com/mlflow/mlflow/pull/17039/files#r2252277312).

We log the spans in batch prior to artifact upload (I should reverse the order).

The UI will still use the artifact as the source of truth, but uploading to the DB will allow us to support filtering traces based on span content. Over time, we can start to migrate away from the artifact as source of truth.

Additionally, as a follow-up, we can flush spans periodically, rather than "all at once" at the end of the trace